sdk/include/reactos/shellutils.h

```diff
@@ -94,6 +94,35 @@ inline BOOL _ROS_FAILED_HELPER(HRESULT hr, const char* expr, const char* filenam
 } /* extern "C" */
 #endif /* defined(__cplusplus) */
 
+static inline UINT
+SHELL_ErrorBoxHelper(HWND hwndOwner, UINT Error)
+{
+    WCHAR buf[400];
+    UINT cch, u32_errstr = 2;
```

Comment:
It is the resource id for the string that says "Error" (and only used when there is no other option if `FormatMessage` fails). This id is the same on Windows.

Reply:
You could do `UINT cch, u32_errstr = IDS_ERROR`
Then at the top of the file you can add:
```
#ifndef IDS_ERROR
#define IDS_ERROR 2
#endif // IDS_ERROR
```
I'm guessing we call it IDS_ERROR in our resources, but double check that.

Reply:
This is shellutils.h, I don't want to pollute globally with macros that may cause conflicts. I will however use a more descriptive name that includes "IDS_ERROR".

Reply:
It will not cause an issue
From Microsoft's MSVC documentation:

>"However, you can redefine the macro with exactly the same definition. Thus, the same definition may appear more than once in a program."

From GNU's GCC documentation

>"If the new definition is effectively the same, the redefinition is silently ignored. This allows, for instance, two different headers to define a common macro. The preprocessor will only complain if the definitions do not match."

https://learn.microsoft.com/en-us/cpp/preprocessor/macros-c-cpp?view=msvc-170
https://gcc.gnu.org/onlinedocs/cpp/Undefining-and-Redefining-Macros.html

Reply:
> It will not cause an issue. It's encouraged in fact.

Where did I say with the same value?!!!

- base/shell/progman/resource_2.h does define IDS_ERROR    0x03
- dll/win32/msi/resource.h does define IDS_ERROR         (IDS_ERROR_BASE + 1)

Reply:
You can use a parameter.

Reply:
Then find the error string resource definition and use that if it isn't `IDS_ERROR` isn't the right definition.
I said this because I was away from home:
> "I'm guessing we call it IDS_ERROR in our resources, but double check that."

Reply:
> Then find the error string resource definition and use that if it isn't `IDS_ERROR`

It is IDS_ERROR, in user32.

You people are impossible to please.

Reply:
To be sure, I added the define I suggested above to my own branch and it didn't cause any conflicts. I don't understand why you're so against it, it's the right thing to do in this instance.

Reply:
By doing it your way, you are saying that MSI.dll is not allowed to use shellutils.h in the future.

And for what purpose? How is a define in any way better than what I'm doing? It is already a slight hack. If FormatMessage fails, user32 is the only acceptable place I can grab a suitable string from.

I'm trying my best to do the right thing here but you are making it very hard. Or do you want the shell to continue its sad state of silently failing file-system operations?

Reply:
Maybe MSI shouldn't redefine IDS_ERROR? MSI is winesynced, it would never use shellutils.h. And if you did end up forking it, you should call something else; it's poorly named.

Reply:
Those are fair points.

I still don't understand how a define is better than something contained inside the function in question.

No matter how we look at it, I'm relying on a user32 implementation detail and that is slightly hacky. But again, it's the best I can do...

Reply:
The define points to why we set it to that value; it corresponds with IDS_ERROR resource in user32. You can add a comment to the top of it too that says `// Pulled from user32 resource IDS_ERROR`

Reply:
I guess that's better, I would dismiss my rejection if you did that. But I still think the define is a better way to do this.